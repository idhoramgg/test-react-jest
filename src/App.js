import React from 'react';
import './App.css';

function App() {
  const [counter, setCounter] = React.useState(0)
  const incrementValue = () => setCounter(counter + 1)
  const limitDecrement = () => counter > 0 ? setCounter(counter - 1) : {}


  return (
    <div className="App">
      <h1 id="hello">Hello world</h1>
      <div id="counter-value">{counter}</div>
      <button onClick={incrementValue} id="increment-btn">Increment</button>
      <button onClick={limitDecrement} id="decrement-btn">Decrement</button>
    </div>
  );
}

export default App;
