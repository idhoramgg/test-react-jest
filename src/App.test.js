import App from './App';
import Header from './Header'

import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer'

describe('Component Testing', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<App/>)
  })

  test('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('render app component', () => {
    expect(wrapper.find('#hello').text()).toContain("Hello world")
  });

  // increment btn
  test('render button with increment', () => {
    expect(wrapper.find('#increment-btn').text()).toBe("Increment")
  })
  // decrement btn
  test('render button with decrement', () => {
    expect(wrapper.find('#decrement-btn').text()).toBe("Decrement")
  })

  test('render counter div', () => {
    expect(wrapper.find('#counter-value').text()).toBe("0")
  })

  // increment
  test('add value counter + 1', () => {
    wrapper.find('#increment-btn').simulate('click');
    expect(wrapper.find('#counter-value').text()).toBe("1")
  })
  test('decerease value counter - 1', () => {
    wrapper.find('#increment-btn').simulate('click');
    expect(wrapper.find('#counter-value').text()).toBe("1")
    wrapper.find('#decrement-btn').simulate('click');
    expect(wrapper.find('#counter-value').text()).toBe("0")
    wrapper.find('#decrement-btn').simulate('click');
    expect(wrapper.find('#counter-value').text()).toBe("0")
  })


})
