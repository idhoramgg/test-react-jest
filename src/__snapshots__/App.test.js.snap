// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Testing matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    id="hello"
  >
    Hello world
  </h1>
  <div
    id="counter-value"
  >
    0
  </div>
  <button
    id="increment-btn"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    id="decrement-btn"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
